akka-projection-core/src/main/scala/akka/projection/scaladsl/Projection.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.scaladsl
+
+import akka.stream.{KillSwitch, KillSwitches, Materializer}
+import akka.stream.scaladsl.{Keep, Sink, Source}
+
+import scala.concurrent.ExecutionContext
+
+case class Projection[Envelope, Event, Offset, Result](sourceProvider: SourceProvider[Offset, Envelope],
+                                                       envelopeExtractor: EnvelopeExtractor[Envelope, Event, Offset],
+                                                       runner: ProjectionRunner[Offset, Result],
+                                                       handler: ProjectionHandler[Event, Result]) {
+
+
+  private var shutdown: Option[KillSwitch] = None
```

Comment:
The goal is to have a Projection running inside a sharded actor. The actor gets a message to start running it and use the `KillSwtich` when shutting down. 

Plus, we need backoff strategy for the source.

We will need to add it on future iterations. I'm keeping it to a minimal, for now, to explore other API aspects first.